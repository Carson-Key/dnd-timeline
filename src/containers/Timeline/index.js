import React, { Component, Fragment } from 'react'
import {
  Button,
  Container,
  Jumbotron
} from 'reactstrap';
import Time from '../Time'
import { capitalizeFirstLetter } from '../../utilities/capitalizeFirstLetter.js'
import { addPossesive } from '../../utilities/addPossesive.js'

class Timeline extends Component {
  render() {
    const playerObject = this.props.playerObject
    const timeline = playerObject.timelines[this.props.timelineName].characters[this.props.characterName].timeline

    return (
      <Fragment>
        <Button
          className="exitButton"
          id="timelineExitButton"
          color="danger"
          onClick={this.props.exitTimeLine}
        >
          exit {capitalizeFirstLetter(this.props.characterName)}
        </Button>
        <Jumbotron fluid>
          <Container fluid>
            <center>
              <h1 className="display-3">
                {addPossesive(capitalizeFirstLetter(this.props.characterName))} Timeline
              </h1>
            </center>
          </Container>
        </Jumbotron>
        <Time
          timeline={timeline.events}
          lessThanYearZero={timeline.lessThanYearZero}
          greaterThanYearZero={timeline.greaterThanYearZero}
        />
      </Fragment>
    )
  }
}

export default Timeline
