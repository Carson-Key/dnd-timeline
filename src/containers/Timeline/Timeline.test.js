import React from "react"
import Timeline from "./index.js"
import { render } from 'enzyme'
import toJson from "enzyme-to-json";

describe('Render Timeline ->', () => {

  test('with minimal props', () => {
    const timeLine = render(
      <Timeline
        playerObject={{
          characters: {
            "cade": {
                "name": "Cade Goodbarrel",
                "characterKey": "cade",
                "timeline": {
                  "events": {}
                }
              }
            }
          }}
        characterName="cade"
      />,
    )

    expect(toJson(timeLine)).toMatchSnapshot()
  })

})
