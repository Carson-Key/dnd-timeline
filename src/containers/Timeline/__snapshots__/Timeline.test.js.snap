// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Timeline -> with minimal props 1`] = `
Array [
  <button
    class="exitButton btn btn-danger"
  >
    exit Cade
  </button>,
  <div
    class="jumbotron jumbotron-fluid"
  >
    <div
      class="container-fluid"
    >
      <center>
        <h1
          class="display-3"
        >
          Cade's Timeline
        </h1>
      </center>
    </div>
  </div>,
  <br />,
]
`;
